#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_patch_pcscf.dpatch by  <orange@ubuntuims1>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch for cfg/pcscf.sh file

@DPATCH@

diff -Nru tmp.orig/cfg/pcscf.sh tmp/cfg/pcscf.sh
--- tmp.orig/cfg/pcscf.sh	2009-04-10 11:58:20.000000000 +0200
+++ tmp/cfg/pcscf.sh	2009-04-10 14:26:25.000000000 +0200
@@ -1,8 +1,18 @@
 #!/bin/bash
 
-./ser_ims/cfg/killser pcscf
-setkey -F
-setkey -FP
+#Variables
+NAME=pcscf
+CONFFILE=/etc/openimscore/$NAME.cfg
+LIBPATH=/usr/lib/ser
+KILLSER=/usr/sbin/killser
+SER=/usr/sbin/ser
 
-LD_LIBRARY_PATH="/usr/local/lib/ser" /opt/OpenIMSCore/ser_ims/ser -f /opt/OpenIMSCore/pcscf.cfg -D -D
+#Verifying files
+[ -x $KILLSER ] && killser $NAME || exit 1
+[ -x setkey ] && setkey -F
+[ -x setkey ] && setkey -FP
+[ -x $SER ] || exit 1
+[ -f $CONFFILE ] || exit 1
 
+LD_LIBRARY_PATH="$LIBPATH" $SER -f $CONFFILE -D -D -D
+exit 0
